src/main/java/org/jenkinsci/test/acceptance/po/PluginManager.java

```diff
@@ -95,23 +97,23 @@ public void checkForUpdates() {
         // The check now button is a form submit (POST) with a redirect to the same page only if the check is successful.
         // We use the button itself to detect when the page has changed, which happens after the refresh has been done
         // And we check for the presence of the button again
-        clickLink("Check now");
+        WebElement checkButton = find(by.link("Check now"));
+        checkButton.click();
         // The wait criteria is: we have left the current page and returned to the same one
-        waitFor(find(by.link("Check now"))).withTimeout(30, TimeUnit.SECONDS).until(webElement -> {
+        waitFor(checkButton).withTimeout(java.time.Duration.of(time.seconds(30), ChronoUnit.MILLIS)).until(webElement -> {
```

Comment:
Not sure what's the difference between this and the previous form. `clickLink` does exactly `find(by.link(..)` (which is what you are doing explicitly now), and then `waitFor(checkButton).withTimeout(java.time.Duration.of(time.seconds(30), ChronoUnit.MILLIS))` is doing exactly the same than `waitFor(find(by.link("Check now"))).withTimeout(30, TimeUnit.SECONDS)`.

Could you explain how this change reduces flakiness?

Reply:
The difference is, in that case, we define the button before clicking it, so the button took into account is the one at the beginning. With this change, we are sure we are waiting for the correct one but not the "Check now" button that reappears after reloading the page

Reply:
@bmunozm, hmm, good catch.